### DISCUSSION TITLE
How to pass parameters to callback functions

### DISCUSSION THREAD

**Discussion by zymouse** [zauberzeug/nicegui#2394] [Q&A]
## Checklist
- [x] Version: nicegui=1.4.11
- [x] Make sure there are no related issues
- [x] I have finished reading the document

## descriptions
As in the above code `on_change=pub_mission_planning_goal`, `pub_mission_planning_goal` makes a callback function, if `pub_mission_planning_goal` is a callback function with parameters, how to pass in the parameters
```py
ui.toggle(self.station_yaml_list_map, on_change=pub_mission_planning_goal)
```

**Comment by falkoschindler** [comment]
Here are three options:

You can create a lambda function
```py
ui.toggle(self.station_yaml_list_map, on_change=lambda: pub_mission_planning_goal(x, y))
```

a regular function
```py
def handle_change():
    pub_mission_planning_goal(x, y)

ui.toggle(self.station_yaml_list_map, on_change=handle_change)
```

or use `partial`:
```py
from functools import partial

ui.toggle(self.station_yaml_list_map, on_change=partial(pub_mission_planning_goal, x, y))
```
